src/service/location/location.service.ts

```diff
@@ -7,18 +7,41 @@ import { firstValueFrom } from 'rxjs';
 import { Address, IPResponse } from './types.location';
 import { CACHE_MANAGER } from '@nestjs/cache-manager';
 import { Cache } from 'cache-manager';
+import {
+  HealthIndicator,
+  HealthIndicatorResult,
+  HttpHealthIndicator,
+} from '@nestjs/terminus';
 
 @Injectable()
-export class LocationService {
+export class LocationService extends HealthIndicator {
   private readonly _resolveAddressCacheKey = 'resolveAddressCache';
   private readonly _locationDetailsCacheKey = 'locationDetailsCache';
   private readonly logger = new Logger(LocationService.name);
 
   constructor(
     private configService: ConfigService,
     private readonly httpService: HttpService,
+    private httpHealthIndicator: HttpHealthIndicator,
     @Inject(CACHE_MANAGER) private readonly cacheManager: Cache,
-  ) {}
+  ) {
+    super();
+  }
+
+  // dedicated, heavily cached healthcheck needed because of ipinfo.io token limits
+  async isHealthy(): Promise<HealthIndicatorResult> {
+    const cachedHealth =
+      await this.cacheManager.get<HealthIndicatorResult>('health');
```

Comment:
I'm not familiar with NestJS dependency injection - is the `cacheManager` instance shared globally, or does it create a separate instance for each injection?
If it's global, then maybe change the key from `health` to `locationHealth`, to avoid potential conflicts?

Reply:
Well Im not sure as well, so Ill just change this :D

Reply:
You need to change it in `await this.cacheManager.set('health', health, 1000 * 60 * 60); // 1 hour` as well

Reply:
Force pushed, evidence destroyed